### PR TITLE
Fixed #24931 -- Made nested RegexURLResolvers populate callback strings

### PR DESCRIPTION
**Pull request by MarkusH**
https://code.djangoproject.com/ticket/24931

### CODE REVIEW

**Comment by timgraham** [comment]
It's always nice to have a test in a place other than a contrib app for functionality in core.

**Comment by MarkusH** [comment]
I agree, the thing is, it's private, undocumented API and only ever used in admindocs. So I'd rather not test this as part of the url resolvers.

In the long term we should either make the underlying API (lookup a view function by its dotted python path) a public API and test it in the url resolvers or remove it from there and make it part of the admindocs.

**Comment by timgraham** [comment]
I don't understand the reasoning. Tests don't need to be restricted to public APIs, do they?

**Comment by MarkusH** [comment]
Hm, not sure where I was going with that either. Anyway, moved the tests to `urlresolver_reverse`.

Previous commit: 5c69ca8a7cf816339fbd73f896c7f2e507b7be11

**Comment by MarkusH** [comment]
This is a regression introduced in 2f16ff5a6cbd71fc6c50e88e4087f3657222e90e

I therefore lean to (additionally) add the test from 5c69ca8.

When this is applied on a24cf217220dca44b7bd5b36ad9c14a96bca486e it passes. Applied to 2f16ff5a6cbd71fc6c50e88e4087f3657222e90e and it fails.
```diff
diff --git a/tests/admin_docs/namespace_urls.py b/tests/admin_docs/namespace_urls.py
new file mode 100644
index 0000000..328b8d2
--- /dev/null
+++ b/tests/admin_docs/namespace_urls.py
@@ -0,0 +1,14 @@
+from django.conf.urls import include, url
+from django.contrib import admin
+
+from . import views
+
+backend_urls = [
+    url(r'^something/$', views.XViewClass.as_view(), name='something'),
+]
+
+urlpatterns = [
+    url(r'^admin/doc/', include('django.contrib.admindocs.urls')),
+    url(r'^admin/', admin.site.urls),
+    url(r'^api/backend/', include(backend_urls, namespace='backend')),
+]
diff --git a/tests/admin_docs/tests.py b/tests/admin_docs/tests.py
index 0826dfc..6e08ddb 100644
--- a/tests/admin_docs/tests.py
+++ b/tests/admin_docs/tests.py
@@ -84,6 +84,12 @@ class AdminDocViewTests(AdminDocsTestCase):
         # View docstring
         self.assertContains(response, 'Base view for admindocs views.')
 
+    @override_settings(ROOT_URLCONF='admin_docs.namespace_urls')
+    def test_namespaced_view_detail(self):
+        url = reverse('django-admindocs-views-detail', args=['admin_docs.views.XViewClass'])
+        response = self.client.get(url)
+        self.assertContains(response, '<h1>admin_docs.views.XViewClass</h1>')
+
     def test_model_index(self):
         response = self.client.get(reverse('django-admindocs-models-index'))
         self.assertContains(
```

File: django/urls/resolvers.py [hunk 1]
```diff
@@ -163,7 +163,7 @@ def __init__(self, regex, urlconf_name, default_kwargs=None, app_name=None, name
         # set of dotted paths to all functions and classes that are used in
         # urlpatterns
         self._callback_strs = set()
-        self._populated = False
+        self._populated = self._populating = False
```
Comment (timgraham) [inline django/urls/resolvers.py]: I guess I'm not strongly -1 but multiple variable assignment on one line is an uncommon style that I'm not sure we want to encourage.

File: tests/urlpatterns_reverse/nested_urls.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+from django.conf.urls import include, url
+from django.views import View
+
+
+def func(request):
```
Comment (timgraham) [inline tests/urlpatterns_reverse/nested_urls.py]: names like view1/view2/View3 might be more readable

**Comment by timgraham** [comment]
Sure, I think including the admindocs test is a good idea.

**Comment by MarkusH** [comment]
Thanks @timgraham

**Pull request merged**